tests/test_certs.py

```diff
@@ -0,0 +1,66 @@
+  "expiration": "2023-09-20T10:02:48Z",
+  "thing_id": "3d49a42f-63fd-491b-9784-adf4b64ef347"
+}
+token =  "eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2OTI2NzM1NTksImlhdCI6MTY5MjYxOTU1OSwiaWRlbnRpdHkiOiJpbnNwaXJpbmdfZ29sZGJlcmdAZW1haWwuY29tIiwiaXNzIjoiY2xpZW50cy5hdXRoIiwic3ViIjoiYzk2ZWQwNmQtNTM5Zi00YjAxLTk2MWUtNTA0MjJmZjQ0OTBkIiwidHlwZSI6ImFjY2VzcyJ9.6J34t9Ts0VmjfsdxeUd66DQihFuB-mwlJ884KwoCjT1FmNr70yPjtIXpOvRjA58ggs-5kTh_ZZh25KtkTipWEg"
+thing_id=  "2544670a-a992-4865-8222-24e5deed7bae"
+cert_id= "22:16:df:60:c2:99:bc:c4:9b:1d:fd:71:5e:e9:07:d9:1b:3c:85:1d"
+
+url= "http://localhost"
+
+def test_issue(requests_mock):
+    requests_mock.register_uri( "POST", url + "/certs", headers={"location": "/certs/" + thing_id}, json=certs, status_code=201)
+    r = s.certs.issue(thing_id=thing_id, valid= "10h", token=token)
+    assert r.error.status == 0
+    assert certs == r.value
+    
+def test_issue_bad_token(requests_mock):
+    requests_mock.register_uri( "POST", url + "/certs", headers={"location": "/certs/" + thing_id}, json=certs, status_code=401)
+    r = s.certs.issue(thing_id=thing_id, valid= "10h", token=token)
+    assert r.error.status == 1
+    assert r.error.message == "Missing or invalid access token provided."
+
+def test_view_by_thing(requests_mock):
+    requests_mock.register_uri( "GET", url + "/serials" + "/" + thing_id, json=certs, status_code=200)
+    r = s.certs.view_by_thing(thing_id=thing_id, token=token)
+    assert r.error.status == 0
+    assert certs == r.value
+    
+def test_view_by_thing_bad_token(requests_mock):
+    requests_mock.register_uri( "GET", url + "/serials" + "/" + thing_id, json=certs, status_code=404)
+    r = s.certs.view_by_thing(thing_id=thing_id, token=token)
+    assert r.error.status == 1
+    assert r.error.message == "Failed to retrieve corresponding certificate."
+    
+def test_view_by_serial(requests_mock):
+    requests_mock.register_uri( "GET", url + "/certs" + "/" + cert_id, json=certs, status_code=200)
+    r = s.certs.view_by_serial(cert_id=cert_id, token=token)
+    assert r.error.status == 0
+    assert certs == r.value
+    
+def test_view_by_serial_bad_token(requests_mock):
+    requests_mock.register_uri( "GET", url + "/certs" + "/" + cert_id, json=certs, status_code=404)
+    r = s.certs.view_by_serial(cert_id=cert_id, token=token)
+    assert r.error.status == 1
+    assert r.error.message == "Failed to retrieve corresponding certificate." 
+
+def test_revoke(requests_mock):
+    requests_mock.register_uri( "DELETE", url + "/certs" + "/" + thing_id, status_code=200)
+    r = s.certs.revoke(thing_id=thing_id, token=token)
+    assert r.error.status == 0
+    
+def test_revoke(requests_mock):

```

Comment:
Can we change this name to test_revoke_bad_response to prevent conflict, each test function needs to have a unique name.